api/src/main/java/com/stormpath/sdk/account/Account.java

```
@@ -560,4 +571,210 @@
+     * @since 1.1.0
+     */
+    AccountLink link(Account otherAccount);
+
+    /**
+     * Links this account to the otherAccount represented by its {@code href}
+     *
+     * <p><b>Immediate Execution:</b> Unlike other Account methods, you do <em>not</em> need to call {@link #save()}
+     * afterwards. This method will persist the changes in the backend immediately.</p>
+     *
+     * @param otherAccountHref the href of the other account that this account will be linked to
+     * @return the new AccountLink resource created reflecting the link between two accounts.
+     * @throws IllegalStateException if no Account matching {@code otherAccountHref} could be found.
+     *
+     * @since 1.1.0
+     */
+    AccountLink link(String otherAccountHref);
+
+    /**
+     * Removes the link between this account and the otherAccount
+     *
+     * <p><b>Immediate Execution:</b> Unlike other Account methods, you do <em>not</em> need to call {@link #save()}
+     * afterwards. This method will persist the changes in the backend immediately.</p>
+     *
+     * @param otherAccount the {@code Account} object from which the account must be unlinked.
+     * @return the AccountLink object that was just deleted
+     * @throws IllegalStateException if this Account is not linked to the otherAccount.
+     *
+     * @since 1.1.0
+     */
+    AccountLink unlink(Account otherAccount);
+
+    /**
+     * Removes the link between this account and the otherAccount represented by its {@code href}
+     *                   .
+     * <p><b>Immediate Execution:</b> Unlike other Account methods, you do <em>not</em> need to call {@link #save()}
+     * afterwards. This method will persist the changes in the backend immediately.</p>
+     *
+     * @param otherAccountHref the href of the other account from which the account must be unlinked.
+     * @return the AccountLink object that was just deleted
```

Shouldn't we just propagate the exception, that we received from the rest layer, if the accounts were not linked, instead of returning null? @lhazlewood Thoughts?